### PR TITLE
Feat/custom badge without hexagon frame

### PR DESCRIPTION
**Pull request by YouQam**
This PR adds an option to show/hide hexagon frame during the process of creating a badge.

### CODE REVIEW

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts [hunk 1]
```diff
@@ -291,25 +312,66 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
...
-            }
-        })
+	ngOnInit() {
+		super.ngOnInit();
+		let that = this;
+		// update badge frame when a category is selected
+		this.badgeClassForm.rawControl.controls['badge_category'].statusChanges.subscribe((res) => {
+			if (this.currentImage) {
+				// Skip update badge icon frame if no-hexagon-frame checkbox is checked
+				if (!this.noHexFrame) {
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts]: Why don't you combine those two if conditions in one `if`?

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts [hunk 2]
```diff
@@ -291,25 +312,66 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
...
+						that.adjustUploadImage(that.badgeClassForm.value);
+					}, 10);
+				}
+			}
+		});
+		// update badge frame when a level is selected
+		this.badgeClassForm.rawControl.controls['badge_level'].statusChanges.subscribe((res) => {
+			if (this.currentImage) {
+				// Skip update badge icon frame if no-hexagon-frame checkbox is checked
+				if (!this.noHexFrame) {
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts]: Same as other comment

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts [hunk 3]
```diff
@@ -605,4 +674,18 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
     openLegend() {
         this.showLegend = true;
     }
+
+	handleCBoxValueChange(e) {
+		this.noHexFrame = !this.noHexFrame;
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts]: Wouldn't it be nicer to bind the value of the checkbox to a property? Then we wouldn't have to rely on the switching to work (it seems there could go wrong a lot of stuff). If the binding is, for some reason, not possible, maybe the value is contained in the event `e`? Or accessible via the native element.

**Review by timber-they — CHANGES_REQUESTED** [review]
I like it, but two things seem improvable to me.

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts (hunk 3 above)
Comment (YouQam) [reply src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts]: Yes, it was not possible to do binding even after creating a `formfield-checkbox`. I updated the impl. to use event.

**Pull request merged**